File: gateway/src/main/java/com/example/gateway/controller/ItemRequestController.java
package com.example.gateway.controller;

import com.example.api.ItemRequestApi;
import com.example.api.dto.ItemRequestDto;
import com.example.gateway.client.ItemRequestClient;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping(path = ItemRequestApi.PATH)
@RequiredArgsConstructor
public class ItemRequestController implements ItemRequestApi {

    private final ItemRequestClient requestClient;

    @Override
    public ItemRequestDto create(ItemRequestDto itemRequestDto, Long userId) {
        return requestClient.create(itemRequestDto, userId);
    }

    @Override
    public ItemRequestDto findById(Long id, Long userId) {
        return requestClient.findById(id, userId);
    }

    @Override
    public List<ItemRequestDto> findAllByRequesterId(Long userId) {
        return requestClient.findAllByRequesterId(userId);
    }

    @Override
    public List<ItemRequestDto> findAllByRequesterIdNot(Long userId, Integer from, Integer size) {
        return requestClient.findAllByRequesterIdNot(userId, from, size);
    }
}

File: gateway/src/main/kotlin/com/example/gateway/controller/ItemRequestController.kt
package com.example.gateway.controller

import com.example.api.ItemRequestApi
import com.example.api.dto.ItemRequestDto
import com.example.gateway.client.ItemRequestClient
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = [ItemRequestApi.PATH])
class ItemRequestController(
    val requestClient: ItemRequestClient
) : ItemRequestApi {

    override fun create(itemRequestDto: ItemRequestDto?, userId: Long?): ItemRequestDto? {
        return requestClient.create(itemRequestDto, userId)
    }

    override fun findById(id: Long?, userId: Long?): ItemRequestDto? {
        return requestClient.findById(id, userId)
    }

    override fun findAllByRequesterId(userId: Long?): List<ItemRequestDto?>? {
        return requestClient.findAllByRequesterId(userId)
    }

    override fun findAllByRequesterIdNot(userId: Long?, from: Int?, size: Int?): List<ItemRequestDto?>? {
        return requestClient.findAllByRequesterIdNot(userId, from, size)
    }
}
